packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -353,57 +353,38 @@ class RenderSliverMainAxisGroup extends RenderSliver
       hasVisualOverflow:
           totalScrollExtent > constraints.remainingPaintExtent || constraints.scrollOffset > 0.0,
     );
+
+    // Update the children's paintOffset based on the direction again, which
+    // must be done after obtaining the `paintExtent`.
+    child = firstChild;
+    while (child != null) {
+      final SliverPhysicalParentData childParentData =
+          child.parentData! as SliverPhysicalParentData;
+      final AxisDirection axisDirection = constraints.axisDirection;
+      final Offset paintOffset = switch (axisDirection) {
```

Comment:
```suggestion
      childParentData.paintOffset = switch (constraints.axisDirection) {
```

Just a nit you can choose to use or ignore. :)